src/Agent/NewRelic/Agent/Core/DataTransport/ConnectionManager.cs

```diff
@@ -112,36 +112,42 @@ private void Connect()
 #if !NETFRAMEWORK // Only available in System.Net.Http
             // Occurs when the agent is unable to connect to APM. The request failed due to an underlying
             // issue such as network connectivity, DNS failure, server certificate validation or timeout.
-            catch (HttpRequestException)
+            catch (HttpRequestException hrex)
             {
+                Log.Info(hrex, "Connection failed.");
```

Comment:
I think this is the one case where we want to just log the exception message rather than the object, which contains the call stack. Even though it's at `Info` level, a lot of customers see a call stack and think "problem".

Reply:
Are you suggesting this change across all of the exceptions that result in a restart? That's probably reasonable.

Reply:
Exactly

Reply:
Refactored to move exception handling to a separate method and only log the message for exceptions that will result in a restart.